lib/solr.js

```diff
@@ -550,19 +549,41 @@ Client.prototype.update = function (data, options, callback) {
     })
     .join('/');
 
-  const params = {
+  const requestOptions = {
     host: this.options.host,
     port: this.options.port,
-    fullPath: fullPath,
-    json: json,
-    secure: this.options.secure,
-    bigint: this.options.bigint,
-    authorization: this.options.authorization,
-    agent: this.options.agent,
-    ipVersion: this.options.ipVersion,
-    request: this.request,
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json; charset=utf-8',
+      'content-length': Buffer.byteLength(json),
+      accept: 'application/json; charset=utf-8',
+    },
+    path: fullPath,
+    family: this.options.ipversion,
```

Comment:
still trying to wrap my head around all these changes. So I see a new field `family` that wasn't here before, and I don't see some of the fields that previously were here. Do all of them come from `this.options.request` now?
sorry for the long delay on reviewing this PR, I have a rather uneasy feeling about it due to the rather opaque changes in it. If you have some time in the nearby future, can we talk them through and push forward with this PR?

Reply:
@rudolfbyker Sorry, it took me a while and adding TypeScript to fully understand what was going here, and I envy your capability to comprehend rather messy code here and actually fix what was broken before. I believe that your change is completely correct here and it should be done like this. Unfortunately, my TypeScript addition and class refactorings caused conflicts in your PR, but it should be easier to make those changes now that we have the safety net.
Would you be open to rebasing this PR? Alternatively I can use it as a basis for a reimplementation for the same refactorings that you were going for.

Reply:
Ah, I'm glad we're on the same page now :) I'm definitely open to rebasing this, or throwing it away completely and using just the spirit of it to help future work. I saw the many other PRs and commits, and will try to make time ASAP to look at those to get up to speed with what you are doing.

Reply:
Appreciate it dearly!
We've been mostly adding types, I did my best to avoid touching request-related code for now.